component/src/main/java/org/wso2/extension/siddhi/io/http/source/util/HttpSourceUtil.java

```diff
@@ -18,76 +18,134 @@
+import org.wso2.transport.http.netty.message.HTTPCarbonMessage;
 
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.ArrayList;
-import java.util.HashSet;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
-
 /**
  * Handles the http source util functions.
  */
 public class HttpSourceUtil {
     private static final Logger log = Logger.getLogger(HttpSourceUtil.class);
+    private static HttpSourceUtil instance = new HttpSourceUtil();
 
     public HttpSourceUtil() {
     }
 
+    public static HttpSourceUtil getInstance() {
+        return instance;
+    }
+
+    public Set<TransportProperty> populateHeaderValidationConfigurations(
+            Map<String, String> serverHeaderValidationList, Set<TransportProperty> transportProperties) {
+        Map<String, TrpPropertyTypes> tryMap = trpPropertyTypeMap();
+        serverHeaderValidationList.forEach((key, value) -> {
+            TransportProperty trpProperty = new TransportProperty();
+            trpProperty.setName(key);
+            switch (tryMap.get(key).name()) {
+                case "BOOLEAN":
+                    trpProperty.setValue(Boolean.valueOf(value));
+                    break;
+                case "STRING":
+                    trpProperty.setValue(value);
+                    break;
+                case "INTEGER":
+                    trpProperty.setValue(Integer.valueOf(value));
+                    break;
+                case "DOUBLE":
+                    trpProperty.setValue(Double.valueOf(value));
+                    break;
+                default:
+                    trpProperty.setValue(value);
+                    break;
+            }
+            transportProperties.add(trpProperty);
+        });
+        return transportProperties;
+    }
+
+    /**
+     * @param parameterList
+     * @return
+     */
+    public List<Parameter> populateParameters(String parameterList) {
+        List<org.wso2.transport.http.netty.config.Parameter> parameters = new ArrayList<>();
+        if (!HttpConstants.EMPTY_STRING.equals(parameterList.trim())) {
+            try {
+                String[] valueList = parameterList.trim().substring(1, parameterList.length() - 1).split("','");
+                Arrays.stream(valueList).forEach(valueEntry ->
+                        {
+                            org.wso2.transport.http.netty.config.Parameter parameter = new Parameter();
+                            parameter.setName(valueEntry.split(":")[0]);
+                            parameter.setValue(valueEntry.split(":")[1]);
+                            parameters.add(parameter);
+                        }
+                );
+
+            } catch (ArrayIndexOutOfBoundsException ex) {
+                log.error("Bootstrap configuration is not in expected format please insert them as 'key1:val1'," +
```

Comment:
Can't we handle this without catching ArrayIndexOutOfBounds

Reply:
Fixed in 6bf095af9d7a4c00220069badb50517cec553139